pkg/plugins/resources/remote/store_test.go

```diff
@@ -425,6 +425,57 @@ var _ = Describe("RemoteStore", func() {
 		})
 	})
 
+	Describe("Delete()", func() {
+		It("should delete the resource", func() {
+			// given
+			name := "tr-1"
+			meshName := "mesh-1"
+			store := setupStore("delete.json", func(req *http.Request) {
+				Expect(req.URL.Path).To(Equal(fmt.Sprintf("/meshes/%s/trafficroutes/%s", meshName, name)))
+			})
+
+			// when
+			resource := sample_core.TrafficRouteResource{}
+			err := store.Delete(context.Background(), &resource, core_store.DeleteByKey("default", name, meshName))
+
+			// then
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("should return error from the api server", func() {
```

Comment:
Let's add a test case to verify that `status 404` is converted into `ResourceNotFound` error